File: backend/api/schemas/criteria.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, ConfigDict


class CriterionCreate(BaseModel):
    label: str
    description: Optional[str] = ""
    sort_order: int = 0
    is_custom: bool = False


class CriterionUpdate(BaseModel):
    label: Optional[str] = None
    description: Optional[str] = None
    sort_order: Optional[int] = None


class CriterionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    goal_id: int
    label: str
    description: Optional[str]
    sort_order: int
    is_custom: bool
    created_at: datetime


class ReorderItem(BaseModel):
    id: int
    sort_order: int


class ReorderRequest(BaseModel):
    items: List[ReorderItem]

File: frontend/src/types/criteria.ts
export interface Criterion {
  id: number;
  goal_id: number;
  label: string;
  description: string | null;
  sort_order: number;
  is_custom: boolean;
  created_at: string;
}

export interface CriterionCreate {
  label: string;
  description?: string;
  sort_order: number;
  is_custom?: boolean;
}

export interface ReorderItem {
  id: number;
  sort_order: number;
}
